Remotion/ObjectBinding/Web.ClientScript/Scripts/BocReferenceValueBase.ts

```diff
@@ -0,0 +1,223 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+
+type ResourceIdentifier = {
+  LoadIconFailedErrorMessage: string,
+  LoadDataFailedErrorMessage: string,
+  SearchStringForDropDownDoesNotMatchRegexMessage: string,
+  NoDataFoundMessage: string,
+  InternetExplorerScreenReaderLabelText: string
+}
+
+type SearchStringValidationInfo = {
+  ValidSearchStringRegex: string,
+  ValidSearchStringForDropDownRegex: string,
+  SearchStringForDropDownDoesNotMatchRegexMessage: string,
+  IgnoreSearchStringForDropDownUponValidInput: string
+}
+
+type CommandInfo = {
+  href: Nullable<string>,
+  target: Nullable<string>,
+  onClick: Nullable<string>,
+  title: Nullable<string>
+}
+
+type IconContext = {
+  businessObjectClass: Nullable<string>,
+  arguments: Nullable<string>
+}
+
+type ErrorHandler = (error: Sys.Net.WebServiceError) => void
+
+type IconInformation = {
+  Url: string,
+  AlternateText: string,
+  ToolTip: string,
+  Width: number,
+  Height: number
+}
+
+class BocReferenceValueBase
```

Comment:
This is from it's intention a base type for the two ReferenceValue-types. @a-ctor is there a way to express this in TypeScript? If so, we might want to consider a follow-up task to improve this code.

Reply:
Class inheritance is done using `extends` instead of `:` but otherwise behaves similarly

Reply:
sounds good. @Stefan-Ilic please make a follow-up ticket with TODO comment for this.